### PR TITLE
[KEYBOARD] add YMDK40 AIR40

### PR DESCRIPTION
**Pull request by dkruyt**
## Description

YMDK40 has a "new" AIR40 board, this will add support for their board, also will use per key RGB matrix.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/ymd40/air40/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    [2] = LAYOUT_ortho_4x12(
+        KC_GRV,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_DEL,
+        KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS,
+        _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_NUHS, KC_NUBS, _______, KC_PGUP, _______,
+        _______, _______, _______, _______, MO(3),   _______, _______, _______, _______, KC_HOME, KC_PGDN, KC_END
+    ),
+    [3] = LAYOUT_ortho_4x12(
+        QK_BOOT, RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI, _______, _______, _______, _______, _______, _______, DB_TOGG,
+        _______, RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD, _______, _______, _______, _______, _______, _______, _______,
+        _______, BL_TOGG, BL_DOWN, BL_UP,   BL_BRTG, _______, _______, _______, _______, _______, _______, _______,
```
Comment (waffle87) [inline keyboards/ymdk/ymd40/air40/keymaps/via/keymap.c:41]:
```suggestion
        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by edela94** [comment]
This is awesome! Would adding this allow VIA to configure the RGB settings as well?

**Comment by dkruyt** [comment]
> This is awesome! Would adding this allow VIA to configure the RGB settings as well?

Yes!!!

File: keyboards/ymdk/ymd40/air40/rules.mk [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+AUDIO_ENABLE = no           # Audio output
+AUDIO_SUPPORTED = no
+KEY_LOCK_ENABLE = no
+
+LAYOUTS = ortho_4x12
+LAYOUTS_HAS_RGB = yes
```
Comment (drashna) [inline keyboards/ymdk/ymd40/air40/rules.mk:24]:
```suggestion
```

File: keyboards/ymdk/ymd40/air40/air40.c [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+void suspend_power_down_kb(void) {
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+
+void suspend_wakeup_init_kb(void) {
+    rgb_matrix_set_suspend_state(false);
+    suspend_wakeup_init_user();
+}
```
[the remark this answers is not in the document]
Comment (drashna) [reply keyboards/ymdk/ymd40/air40/air40.c:52]: These need to be removed, as they're deprecated.

File: keyboards/ymdk/ymd40/air40/config.h [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+#define DIODE_DIRECTION COL2ROW
+
+#define RGB_DI_PIN E2
+#define RGB_MATRIX_LED_COUNT 51
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_KEYPRESSES // reacts to keypresses
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 130 // Limit to vendor-recommended value
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED // turn off effects when suspended
+#    define RGB_MATRIX_MAXIMUM_BRIGHTNESS 130 // Limit to vendor-recommended value
```
Comment (drashna) [inline keyboards/ymdk/ymd40/air40/config.h:46]: Don't need this twice ;)
```suggestion
```

File: keyboards/ymdk/ymd40/air40/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ortho_4x12(
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSPC,
+        KC_ESC,  KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+        KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_UP,   KC_ENT,
+        KC_LCTL, KC_LALT, KC_LGUI, KC_PIPE, MO(1),   KC_SPC,  KC_SPC,  MO(2),   KC_SLSH, KC_LEFT, KC_DOWN, KC_RGHT
```
Comment (drashna) [inline keyboards/ymdk/ymd40/air40/keymaps/via/keymap.c:24]: Just a heads up, via has `FN_MO13` and `FN_MO23` which do the tri layer stuff.

**Comment by dkruyt** [comment]
Tested and verified by someone else... https://www.reddit.com/r/olkb/comments/zuzvf4/ymdk_air40_leds_only_working_for_8_keys/j1mjdw7/?context=3

**Review by drashna — APPROVED** [review]

File: keyboards/ymdk/ymd40/air40/rules.mk [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+RGB_MATRIX_ENABLE = yes
```
Comment (drashna) [inline keyboards/ymdk/ymd40/air40/rules.mk:17]:
```suggestion
RGBLIGHT_ENABLE = no
RGBLIGHT_SUPPORTED = no
RGB_MATRIX_ENABLE = yes
```

**Review by drashna — COMMENTED** [review]
This will also help with compilation/CI errors.

**Review by fauxpark — APPROVED** [review]

**Pull request merged**